multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -24,20 +31,672 @@ use node_lib::database::db_initializer::{
+    payment_thresholds_all_nodes: PaymentThresholds,
+    cons_node_transaction_fee_agreed_unit_price_opt: Option<u64>,
+
+    exp_final_cons_node_transaction_fee_balance_minor: u128,
+    exp_final_cons_node_service_fee_balance_minor: u128,
+    exp_final_service_fee_balance_serv_node_1_minor: u128,
+    exp_final_service_fee_balance_serv_node_2_minor: u128,
+    exp_final_service_fee_balance_serv_node_3_minor: u128,
+}
+
+enum NodeByRole {
+    ConsNode,
+    ServNode1,
+    ServNode2,
+    ServNode3,
+}
+
+struct SimpleSimulatedDebts {
+    owed_to_serving_node_1_minor: u128,
+    owed_to_serving_node_2_minor: u128,
+    owed_to_serving_node_3_minor: u128,
+}
+
+struct FullySpecifiedSimulatedDebts {
+    owed_to_serving_node_1: AccountedDebt,
+    owed_to_serving_node_2: AccountedDebt,
+    owed_to_serving_node_3: AccountedDebt,
+}
+
+struct AccountedDebt {
+    balance_minor: u128,
+    age_s: u64,
+}
+
+impl TestInputsOutputsConfig {
+    fn port(&self, requested: NodeByRole) -> Option<u16> {
+        self.ui_ports_opt.as_ref().map(|ports| match requested {
+            NodeByRole::ConsNode => ports.consuming_node,
+            NodeByRole::ServNode1 => ports.serving_node_1,
+            NodeByRole::ServNode2 => ports.serving_node_2,
+            NodeByRole::ServNode3 => ports.serving_node_3,
+        })
+    }
+
+    fn debt_size(&self, requested: NodeByRole) -> u128 {
+        match self.debts_config.as_ref() {
+            Either::Left(simple_config) => match requested {
+                NodeByRole::ServNode1 => simple_config.owed_to_serving_node_1_minor,
+                NodeByRole::ServNode2 => simple_config.owed_to_serving_node_2_minor,
+                NodeByRole::ServNode3 => simple_config.owed_to_serving_node_3_minor,
+                NodeByRole::ConsNode => panic!(
+                    "Version simple: These configs are \
+                serve to set owed to the consuming node, while that one should not be here."
+                ),
+            },
+            Either::Right(fully_specified) => match requested {
+                NodeByRole::ServNode1 => fully_specified.owed_to_serving_node_1.balance_minor,
+                NodeByRole::ServNode2 => fully_specified.owed_to_serving_node_2.balance_minor,
+                NodeByRole::ServNode3 => fully_specified.owed_to_serving_node_3.balance_minor,
+                NodeByRole::ConsNode => panic!(
+                    "Version fully specified: These configs \
+                are serve to set owed to the consuming node, while that one should not \
+                be here."
+                ),
+            },
+        }
+    }
+}
+
+struct Ports {
+    consuming_node: u16,
+    serving_node_1: u16,
+    serving_node_2: u16,
+    serving_node_3: u16,
+}
+
+struct ServingNodeAttributes {
+    name: String,
+    index: usize,
+    config: NodeStartupConfig,
+}
+
+fn test_body<StimulateConsumingNodePayments, StartServingNodesAndLetThemPerformReceivablesCheck>(
```

Comment:
The only concern that I have here is the length of this function, which makes it difficult to identify the important stuff from less important items.

A personal preference is:
- Using refactored functions
- Structs in place of tuples (so that things are explicit)